app/src/main/java/org/onionshare/android/ShareManager.kt

```diff
@@ -125,6 +129,8 @@ class ShareManager @Inject constructor(
             // TODO check if this always works as expected
             startSharingJob?.cancelAndJoin()
         }
+        // the ErrorAddingFile state is transient and needs manual reset to not persist
+        if (shareState.value is ShareUiState.ErrorAddingFile) fileManager.resetError()
```

Comment:
We have to leave the error state somehow. As our sharing state depends on the managers and `shouldStop`, we need to modify the state of one of the managers for this. Otherwise, changing `shouldStop` in the next line emits a new state keeping us in an error state.

Reply:
The comment meant to say that the error state should be transient, but that doesn't magically happen, so we need to reset it manually for transient behavior.